examples/cc3200/xively_firmware_updates/SFT/xi_sft_client.c

```diff
@@ -0,0 +1,1336 @@
+                                       const char* details_str,
+                                       const int value_int,
+                                       const sft_status_t status );
+void xi_publish_device_log_format_int_int( xi_context_handle_t in_context_handle,
+                                           const char* message_str,
+                                           const char* details_str,
+                                           const int value_int1,
+                                           const int value_int2,
+                                           const sft_status_t status );
+void xi_publish_device_log_format_str( xi_context_handle_t in_context_handle,
+                                       const char* message_str,
+                                       const char* details_str,
+                                       const char* value_str,
+                                       const sft_status_t status );
+void xi_publish_device_log_format_str_str( xi_context_handle_t in_context_handle,
+                                           const char* message_str,
+                                           const char* details_str,
+                                           const char* value_str1,
+                                           const char* value_str2,
+                                           const sft_status_t status );
+
+/* Utility Functions */
+void print_hash( unsigned char hash[] );
+void verify_sha256( xi_context_handle_t in_context_handle );
+
+
+/*
+ *  Secure File Transfer Client Source Code
+ */
+
+
+/* @brief Secure File Transfer client initialization.
+ *
+ * Formats topics strings to be used for SFT, subscribes to the service and checks for any
+ * pending
+ * commit from the previous boot.
+ */
+int xi_sft_init( xi_context_handle_t in_context_handle,
+                 char* account_id,
+                 char* device_id )
+{
+    if ( NULL == account_id || NULL == device_id )
+    {
+        return -1;
+    }
+
+    /* xi_stopic is the MQTT topic we PUB to to send messages to the broker. */
+    snprintf( xi_stopic, sizeof( xi_stopic ), "xi/ctrl/v1/%s/svc", device_id );
+
+    /* xi_ctopic is the MQTT topic we SUBSCRIBE to to receive messages from the broker. */
+    snprintf( xi_ctopic, sizeof( xi_ctopic ), "xi/ctrl/v1/%s/cln", device_id );
+
+    /* xi_logtopic is the MQTT topic we PUB to to add log messages. */
+    snprintf( xi_logtopic, sizeof( xi_logtopic ), "xi/blue/v1/%s/d/%s/_log", account_id,
+              device_id );
+
+    /* Accept the latest firmware if we get here. */
+    if ( sl_extlib_FlcIsPendingCommit() )
+    {
+        printf( "\n********** Committing this revision:\"%s\" **********\n",
+                firmware_revision );
+        sl_extlib_FlcCommit( FLC_COMMITED );
+        xi_publish_device_log( in_context_handle, "Firmware Committed",
+                               "Boot loader will now boot to new image", sft_status_ok );
```

Comment:
To be more clear I am missing a "from now on" like statement from this expression.